utils/utils.go

```diff
@@ -489,7 +539,9 @@ func GetOpenPort(addr string, port uint32, state *State, sshConn *SSHConnection)
 			if viper.GetBool("bind-random-ports") || !first || err != nil {
 				reportUnavailable(true)
 
-				if viper.GetString("port-bind-range") != "" {
+				if viper.GetBool("use-ports-from-keys") {
```

Comment:
Why would we want to do this?

Reply:
This is basically a security feature. I wanted to prevent the situation when one might use the same private key for multiple connections which would result in assigning random ports. I do not want to allow port forwarding in such a case.

Reply:
But if I redo this slightly, I can just add an optional limitation for GetRandomPortInRange (which would make sense when I implement permitlisten option correctly) and then this part can be untouched